source/apiVolontaria/volunteer/permissions.py

```diff
@@ -1,21 +1,70 @@
 from rest_framework import permissions
 
 
-class IsOwnerOrReadOnly(permissions.BasePermission):
+class EventsPermissions(permissions.BasePermission):
     """
     Custom permission to only allow owners of an object to edit it.
-
-    This is used for Participation only for now.
     """
 
+    @staticmethod
+    def if_can_do_actions(request, obj):
+        if_django_perm = False
+
+        if request.method == 'PATCH':
+            if_django_perm = request.user.has_perm(
+                'volunteer.change_event')
+
+        elif request.method == 'DELETE':
+            if_django_perm = request.user.has_perm(
+                'volunteer.delete_event')
+
+        is_manager = request.user in obj.cell.managers.all()
+
+        return if_django_perm or is_manager
+
     def has_object_permission(self, request, view, obj):
         # Read permissions are allowed to any request,
         # so we'll always allow GET, HEAD or OPTIONS requests.
         if request.method in permissions.SAFE_METHODS:
             return True
 
-        # Write permissions are only allowed to the owner of the snippet.
-        if obj.user == request.user or request.user.is_superuser:
+        # Write permissions are only allowed to the owner of the snippet
+        # or to a Cell Manager.
+        can_edit = self.if_can_do_actions(request, obj)
+
+        return can_edit
```

Comment:
Why create a new function in place of write this code here ?
Actually if we change the content of the function the comment of this one will be outdated and developer can't know they need to change it.

Reply:
I have follow the structure of the ParticipationsPermissions.  In this cas, I have used the function like this in the view to know which serializer to return

```
def get_serializer_class(self):
        # Check Permissions
        can_edit = ParticipationsPermissions.if_can_do_actions(
            self.request, self.get_object()
        )

        # If authenticated user is admin or is a cell manager
        if can_edit:
            return serializers.ParticipationAdminSerializer
        else:
            return serializers.ParticipationBasicSerializer
```

The EventsPermissions if_can_do_actions is not call externally but I have choose this for uniformity.

Reply:
It's exactly the same check as in the permissions.  That why I have make the check as a function in the Permissions and call it in the Permission and here

Reply:
Hum, for me it can be a problem. We have some differences because it's not because i can do an action than i can do it with an other serializer.

@francisjeanneau can you take a look on this review since you check a lot of new thing in DRF ?